### PR TITLE
fix when mouse leaves the avatar or menu area the menu should be closed

### LINKED ISSUE #11015: [UI] When you hover on your avatar and the mouse leaves the area the menu should be closed.
### Desired Behavior
[Screencast from 2024-05-22 19-34-56.webm](https://github.com/meshery/meshery/assets/99207534/92ab07b0-94b8-4ea3-a90f-fb83543d3bc6)

### Suggested Behavior
When the mouse leaves the area then sub-menu should close .

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by EleisonC**
**Notes for Reviewers**

This PR fixes #11015

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.
 
[Hover.webm](https://github.com/meshery/meshery/assets/30932534/1ad86e69-18be-4704-b66b-4c87c3d4de55)

### CODE REVIEW

**Comment by EleisonC** [comment]
@Jougan-0 @leecalcote

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

File: ui/components/User.js [hunk 1]
```diff
@@ -173,11 +177,17 @@ const User = (props) => {
 
   const { color, iconButtonClassName, avatarClassName, classes } = props;
 
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (EleisonC) [reply ui/components/User.js:180]: Hey, @aabidsofi19 could you point me to the library or popper event that I would use to achieve this without timeouts?

[the remark this answers is not in the document]
Comment (leecalcote) [reply ui/components/User.js:180]: Checking back in...

[the remark this answers is not in the document]
Comment (EleisonC) [reply ui/components/User.js]: Hey @leecalcote I was able to solve it with out the timeout am just dealing with the merge conflicts from signing off

**Comment by l5io** [comment]
🚀 Preview for commit 02e3419ef683e0339f3dad01d423f43006542127 at: https://6669a265c556462db01a1dce--meshery-docs-preview.netlify.app

**Review by leecalcote — CHANGES_REQUESTED** [review]
Over 2,000 file changes included here. This PR should contain only changes pertinent to the issue it’s fixing. Commits from other people for other issues should not be included.

**Comment by l5io** [comment]
🚀 Preview for commit 692fe233156f3c0f4e2554ced5320fb519214092 at: https://6669a9e8fd5589343c24dda4--meshery-docs-preview.netlify.app

**Pull request closed without merging**